link_linux.go

```diff
@@ -1001,12 +1004,20 @@ func (h *Handle) linkModify(link Link, flags int) error {
 				cleanupFds(fds)
 				return fmt.Errorf("Tuntap IOCTL TUNSETIFF failed [%d], errno %v", i, errno)
 			}
+			// 1) we only care for the name of the first tap in the multi queue set
+			// 2) if the original name was empty, the localReq has now the actual name
+			if i == 0 {
```

Comment:
is it worth adding also: `base.Name == ""`?

Reply:
imo, if the creation of the interface is successful, then the localReq struct always has the final say on what the name actually is, regardless what was passed into the syscall.

Reply:
TBH i think that we were doing that up to this change incorrectly because we should always read name returned by kernel, not only in case of empty name, but also in case of using name templates.
I did few years ago similar code in https://github.com/rkt/rkt/blob/master/networking/tuntap/tuntap.go - where you can see `tap%d` as name, where kernel in `%d` puts a next available number.

Without this commit we have simply broken handling not only empty names, but also name templates.
So this commit fixes both things in the same time, even if that was not planned by its autor ;)
(still it's worth to note that in e.g. PR description)

Reply:
@rschmied more than in the PR description we should add a note in the method itself. PR description are anyway of no use during development.
Would be great if you can add a comment on the method itself.